packages/create/src/cli/add/change-data-source.ts

```diff
@@ -0,0 +1,146 @@
+        const manifest = await getManifest(basePath, defaults, logger, variant);
+        const dataSources = manifest['sap.app'].dataSources;
+        if (!dataSources) {
+            throw new Error('No data sources found in the manifest');
+        }
+        const answers = await promptYUIQuestions(getPromptsForChangeDataSource(dataSources), false);
+
+        const fs = await generateChange<ChangeType.CHANGE_DATA_SOURCE>(basePath, ChangeType.CHANGE_DATA_SOURCE, {
+            variant,
+            dataSources,
+            answers
+        });
+
+        if (!simulate) {
+            await new Promise((resolve) => fs.commit(resolve));
+        } else {
+            await traceChanges(fs);
+        }
+    } catch (error) {
+        logger.error(error.message);
+        if (error.code === 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY' && !defaults.ignoreCertErrors) {
+            logger.error('If you are using a self-signed certificate, please use the --ignore-cert-errors flag.');
+            const confirm = await prompts([
+                {
+                    type: 'confirm',
+                    name: 'ignoreCertErrors',
+                    message: 'Do you want to ignore certificate errors?'
+                }
+            ]);
+            defaults.ignoreCertErrors = confirm.ignoreCertErrors;
+            await changeDataSource(basePath, defaults, simulate);
+            return;
+        }
+        if (error.response?.status === 401 && loginAttempts) {
+            loginAttempts--;
+            logger.error(`Authentication failed. Please check your credentials. Login attempts left: ${loginAttempts}`);
+            await changeDataSource(basePath, defaults, simulate);
+            return;
+        }
+        logger.debug(error);
+    }
+}
+
+/**
+ *
+ * @param basePath
+ * @param defaults
+ * @param logger
+ * @param variant
+ * @returns
+ */
+async function getManifest(basePath: string, defaults: PromptDefaults, logger: ToolsLogger, variant: any) {
+    const ui5Config = await UI5Config.newInstance(readFileSync(join(basePath, 'ui5.yaml'), 'utf-8'));
+    const { destination, url, client } =
+        ui5Config.findCustomMiddleware<PreviewConfigAdp>('fiori-tools-preview')?.configuration?.adp?.target ?? {};
+
+    let target;
+    if (destination) {
+        target = { destination };
+    } else if (url) {
+        target = { url, client };
+    } else {
+        throw new Error('No system configuration found in ui5.yaml');
+    }
+
+    const provider = await createAbapServiceProvider(
+        target,
+        {
+            ignoreCertErrors: defaults.ignoreCertErrors
+        },
+        true,
+        logger
+    );
+
+    const appIndexService = provider.getAppIndex();
+    const manifestUrl = await appIndexService.getManifestUrl(variant.reference);
+    const lrepService = provider.getLayeredRepository();
+    const manifest = await lrepService.getManifest(manifestUrl);
+    return manifest;
+}
+
+/**
+ *
```

Comment:
Missing JSDoc description and returns, also add return type

Reply:
Fixed with: 01ea9ff8d5b8b50622d2cb9c299a9560cded6c15